src/components/RoiCalculator.vue

```diff
@@ -0,0 +1,432 @@
+<script setup lang="ts" name="RoiCalculator">
+import { KlayIconSwitch } from '~klay-icons'
+import { SModal } from '@soramitsu-ui/ui'
+import BigNumber from 'bignumber.js'
+import { RoiType, Tab } from '@/types'
+
+const Period = {
+  d1: '1D',
+  d7: '7D',
+  d14: '14D',
+  d30: '30D',
+  y1: '1Y',
+  y5: '5Y'
+} as const
+
+type Period = typeof Period[keyof typeof Period]
+
+const StakeTabs = {
+  d1: Period.d1,
+  d7: Period.d7,
+  d30: Period.d30,
+  y1: Period.y1,
+  y5: Period.y5
+} as const
+
+type StakeTabs = typeof StakeTabs[keyof typeof StakeTabs]
+
+const CompoundingTabs = {
+  d1: Period.d1,
+  d7: Period.d7,
+  d14: Period.d14,
+  d30: Period.d30
+} as const
+
+type CompoundingTabs = typeof CompoundingTabs[keyof typeof CompoundingTabs]
+
+const StakeUnits = {
+  tokens: 'tokens',
+  USD: 'USD',
+} as const
+
+type StakeUnits = typeof StakeUnits[keyof typeof StakeUnits]
```

Comment:
I'd like to move it out from the component

Reply:
Done